src/compas_fab/robots/reachability_map/reachability_map.py

```diff
@@ -0,0 +1,157 @@
+from compas.data import Data
+from compas.geometry import Frame
+from compas.robots import Configuration
+from compas.geometry import argmax
+
+
+class ReachabilityMap(Data):
+    """The ReachabilityMap describes the reachability of a robot.
+
+    The ReachabilityMap describes the reachability of a robot at certain frames,
+    with valid IK solutions at these frames. The map only makes sense to be
+    calculated with analytic inverse kinematic solvers, as they include all
+    possible solutions to be found.
+
+    Attributes
+    ----------
+    frames : list of :class:`compas.geometry.Frame` or list of list of :class:`compas.geometry.Frame`
+        The frames at which the IK solutions are calculated, either a 1D list or
+        a 2D list.
+    configurations : list of list of :class:`compas.robots.Configuration` or list of list of list of :class:`compas.robots.Configuration`
+        The configurations at the frames, either a 2D list or a 3D list,
+        depending on the dimension of the frames respectively.
+    score : list of int
+        The number of solutions per frame (1D) or per frame list (2D)
+    points : list of :class:`compas.geometry.Point`
+        The points per frame (1D) or per frame list (2D)
+    shape : tuple of int
+        The shape of the frame array
+
+
+    Links
+    -----
+    http://wiki.ros.org/reuleaux
+    """
+
+    def __init__(self, frames=None, configurations=None, name=None):
+        super(ReachabilityMap, self).__init__(name)
+        self.frames = frames or []  # 1D or 2D
+        self.configurations = configurations or []  # 2D or 3D
```

Comment:
We spoke about this and perhaps it is too difficult to change, but the fact that these two properties can take different shapes leads to code that is more complex than needed. Isn't it possible to always work assuming we are in the higher dimensionality (2D for frames, 3D for configurations) and let the lower ones simply wrap one extra time all the data?

Reply:
you are right, I changed it now to always be 2D frames with 3D configurations.
i wanted actually to accomodate for XD generators, but maybe that only exists in rare cases.